src/sqlite3/schema_processor.cr

```diff
@@ -131,6 +131,10 @@ module Jennifer
             foreign_keys.each do |key|
               io << ","
               io << "FOREIGN KEY (" << key.column << ") REFERENCES " << key.to_table << "(" << key.primary_key << ")"
+              on_update = key.on_update || "RESTRICT"
```

Comment:
I guess it would be better to add nothing to query if `on_update` and `on_delete` is `nil`. Also I', thinking about adding enum (or just case-when statement somewhere) to `ForeignKey` to prevent developer from inputting value that would be inserted directly to query. What do you think?

Reply:
PostgreSQL, MySQL and MS SQL and enforce `REFERENCES` by default. Sqlite is an outlier in terms of it's default and meaning of `NO ACTION`.  For Sqlite `NO ACTION` does nothing.  I don't understand the use case of specifying a constraint that's not used.

Here's how a few SQL implementations handle foreign key constraints.

| SQL        | NO ACTION       | RESTRICT  | Notes |
| ------------- |:-------------:| -----:| --- |
| Sequel      | **Ignored**  | Enforced |
| MySQL      | Enforced | Enforced | Identical since deferrable isn't supported.
| PostgresQL   |  Enforced| Enforced | NO ACTION allows deferrable while RESTRICT is always immediate.
| MS SQL | Enforced | Doesn't exist |

Should the behavior of `t.references` be uniform or db specific?

Should a `t.references` be enforced by default?  I think so.  That's why I put them in my migrations.  To do so `RESTRICT` must be set for Sqlite making `t.references` behavior uniform when using different databases.

Reply:
Agree - we could do `RESTRICT` a uniform behavior :+1: